### PR TITLE
cli: propagate environment variables to apps

### PR DESCRIPTION
**Pull request by eandre**
This adds support for propagating environment variables
from the CLI to the running Encore application, for both
'encore run' and 'encore test'.

It additionally takes extra care to unset the Encore-internal
environment variables as part of the service initialization.

Thanks @gmlewis for the suggestion and feedback!

### CODE REVIEW

**Review by gmlewis — COMMENTED** [review]
Thank you, @eandre !
After reading through this, I'm still a little fuzzy on how the target Encore app gets its environment variables.  I don't see them set anywhere except `Environ: os.Environ()`, and the example usage I see calls the `/echo.Env` endpoint... but that almost looks like a test function.

Is it possible to add an example from an end-user's perspective, either in a README.md or in a simple example app?  (Or maybe I missed it?)

**Review by DomBlack — APPROVED** [review]
This looks good to me.

**Review by gmlewis — APPROVED** [review]
LGTM and thanks again!

File: cli/daemon/run/run.go [hunk 1]
```diff
@@ -358,10 +364,10 @@ func (r *Run) startProc(params *startProcParams) (p *Proc, err error) {
 	runtimeJSON, _ := json.Marshal(runtimeCfg)
 
 	cmd := exec.Command(params.BinPath)
-	cmd.Env = []string{
-		"ENCORE_RUNTIME_CONFIG=" + base64.RawURLEncoding.EncodeToString(runtimeJSON),
-		"ENCORE_APP_SECRETS=" + encodeSecretsEnv(params.Secrets),
-	}
+	cmd.Env = append(params.Environ,
```
[the remark this answers is not in the document]
Comment (eandre) [reply cli/daemon/run/run.go:367]: Yeah exactly, that's the idea :)

**Pull request merged**